Assets/Scripts/Framework/Train/GearboxUIUpdater.cs

```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+
+namespace DerailedDeliveries.Framework.Train
+{
+    /// <summary>
+    /// A class responsible for updating the gearbox UI.
+    /// </summary>
+    public class GearboxUIUpdater : MonoBehaviour
+    {
+        [SerializeField]
+        private Transform _handle;
+
+        [SerializeField]
+        private float[] _handlePositions;
+
+        private void OnEnable()
+        {
+            TrainEngine.Instance.OnSpeedStateChanged += HandleSpeedStateChanged;
+
+            HandleSpeedStateChanged(TrainEngine.Instance.CurrentSpeedIndex);
+        }
+
+        private void OnDisable()
+        {
+            if(TrainEngine.Instance == null)
+                return;
+
+            TrainEngine.Instance.OnSpeedStateChanged -= HandleSpeedStateChanged;
+        }
+
+        private void HandleSpeedStateChanged(int newSpeedState) =>
```

Comment:
```suggestion
        private void HandleSpeedStateChanged(int newSpeedState)
        {
            float _newHandlePosition = _handlePositions[newSpeedState + TrainEngine.SPEED_VALUES_COUNT];
            _handle.rotation = Quaternion.Euler(0, 0, _newHandlePosition);
        }
```

Line is one char to long 😁 This would also be a bit more readable in my opinion.